st2common/bin/migrations/v3.5/st2-migrate-db-dict-field-values

```diff
@@ -0,0 +1,306 @@
+"""
+
+import sys
+import time
+import traceback
+
+from st2common import config
+from st2common.service_setup import db_setup
+from st2common.service_setup import db_teardown
+from st2common.models.db.execution import ActionExecutionDB
+from st2common.models.db.workflow import WorkflowExecutionDB
+from st2common.models.db.workflow import TaskExecutionDB
+from st2common.models.db.trigger import TriggerInstanceDB
+from st2common.persistence.execution import ActionExecution
+from st2common.persistence.liveaction import LiveAction
+from st2common.persistence.workflow import WorkflowExecution
+from st2common.persistence.workflow import TaskExecution
+from st2common.persistence.trigger import TriggerInstance
+from st2common.constants.action import LIVEACTION_COMPLETED_STATES
+from st2common.constants.triggers import TRIGGER_INSTANCE_COMPLETED_STATES
+
+
+def migrate_executions() -> None:
+    """
+    Perform migrations for execution related objects (ActionExecutionDB, LiveActionDB).
+    """
+    print("Migrating execution objects")
+
+    # NOTE: We first only retrieve the IDs because there could be a lot of objects in the database
+    # and this could result in massive ram use. Technically, mongoengine loads querysets lazily,
+    # but this is not always the case so it's better to first retrieve all the IDs and then retrieve
+    # objects one by one.
+    # Keep in mind we need to use ModelClass.objects and not PersistanceClass.query() so .only()
+    # works correctly - with PersistanceClass.query().only() all the fields will still be retrieved.
+
+    # 1. Migrate ActionExecutionDB objects
+    execution_dbs = ActionExecutionDB.objects(
+        __raw__={
+            "result": {
+                "$not": {
+                    "$type": "binData",
+                },
+            },
+            "status": {
+                "$in": LIVEACTION_COMPLETED_STATES,
+            },
+        },
+    ).only("id")
+    execution_ids = [str(execution_db.id) for execution_db in execution_dbs]
+
+    if not execution_ids:
+        print("Found no ActionExecutionDB objects to migrate.")
+        return None
+
+    print("Will migrate %s ActionExecutionDB objects" % (len(execution_ids)))
+    print("")
+
+    for execution_id in execution_ids:
+        execution_db = ActionExecution.get_by_id(execution_id)
+        print("Migrating ActionExecutionDB with id %s" % (execution_id))
+
+        # This is a bit of a "hack", but it's the easiest way to tell mongoengine that a specific
+        # field has been updated and should be saved. If we don't do, nothing will be re-saved on
+        # .save() call due to mongoengine only trying to save what has changed to make it more
+        # efficient instead of always re-saving the whole object.
+        execution_db._mark_as_changed("result")
+        execution_db._mark_as_changed("result_size")
+
+        #  print(getattr(execution_db, "_changed_fields", []))
```

Comment:
Obsolete comment? Although it may remain as it may be useful for debugging in the future.

Reply:
Yeah, I mostly left it there for debugging purposes, will add a comment.